lib/meandro/rules/unused_record_fields.ex

```diff
@@ -33,64 +33,98 @@ defmodule Meandro.Rule.UnusedRecordFields do
   end
 
   defp analyze_module(file, ast, files_and_asts) do
-    {_, acc} = Macro.prewalk(ast, %{current_module: nil, records: []}, &collect_record_info/2)
+    {_, acc} =
+      Macro.prewalk(
+        ast,
+        %{tracking_mode: :all, current_module: nil, records: []},
+        &collect_record_info/2
+      )
 
-    for {module, name, scope, _num_fields, unused_fields, line} = record <-
+    for {module, name, scope, _field_count, unused_fields, line} = record <-
           Enum.reverse(acc[:records]),
         unused_field <- unused_fields,
-        unused_fields != [] do
+        is_unused?(record, files_and_asts) do
       camel_name = name |> Atom.to_string() |> Macro.camelize()
 
-      cond do
-        scope == :private ->
-          # we don't need to check the other files because it's private
-          %Meandro.Rule{
-            file: file,
-            line: line,
-            module: module,
-            text:
-              "Private record :#{name} (#{camel_name}) has an unused field in the module: #{unused_field}",
-            pattern: {name, unused_field}
-          }
-
-        is_unused?(record, files_and_asts) ->
-          %Meandro.Rule{
-            file: file,
-            line: line,
-            module: module,
-            text:
-              "Public record :#{name} (#{camel_name}) has an unused field in the module: #{unused_field}",
-            pattern: {name, unused_field}
-          }
+      if scope == :private do
```

Comment:
What's the difference between the two branches of this `if`? 🤔

Reply:
Private records (those defined with `Record.definerecordp/3`) cannot be used outside the module they are defined, and thus should not be looked for in other files.

Reply:
Right… but in the code… what's the difference here?